browser/brave_rewards/rewards_panel_coordinator.h

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_BRAVE_REWARDS_REWARDS_PANEL_COORDINATOR_H_
+#define BRAVE_BROWSER_BRAVE_REWARDS_REWARDS_PANEL_COORDINATOR_H_
+
+#include <memory>
+#include <string>
+
+#include "base/observer_list.h"
+#include "base/scoped_observation.h"
+#include "brave/components/brave_rewards/common/brave_rewards_panel.mojom.h"
+#include "chrome/browser/ui/browser_user_data.h"
+#include "url/gurl.h"
+
+namespace brave_rewards {
+
+class RewardsPanelCoordinator
+    : public BrowserUserData<RewardsPanelCoordinator> {
+ public:
+  explicit RewardsPanelCoordinator(Browser* browser);
+
+  RewardsPanelCoordinator(const RewardsPanelCoordinator&) = delete;
+  RewardsPanelCoordinator& operator=(const RewardsPanelCoordinator&) = delete;
+
+  ~RewardsPanelCoordinator() override;
+
+  static bool IsRewardsPanelURLForTesting(const GURL& url);
+
+  static void RegisterForBrowser(Browser* browser);
+
+  // Opens the Rewards panel with the default view.
+  bool OpenRewardsPanel();
+
+  // Displays the Rewards onboarding tour in the Rewards panel.
+  bool ShowRewardsTour();
+
+  // Displays a grant captcha for the specified grant in the Rewards panel.
+  bool ShowGrantCaptcha(const std::string& grant_id);
+
+  // Opens the Rewards panel in order to display the currently scheduled
+  // adaptive captcha for the user.
+  bool ShowAdaptiveCaptcha();
+
+  // Opens the Rewards panel in order to display the Brave Talk Rewards opt-in.
+  bool ShowBraveTalkOptIn();
+
+  class Observer : public base::CheckedObserver {
+   public:
+    // Called when an application component requests that the Rewards panel be
+    // opened. The arguments provided by |OpenRewardsPanel| must be retrieved
+    // using |TakePanelArguments|.
+    virtual void OnRewardsPanelRequested(const mojom::RewardsPanelArgs& args) {}
+
+    // Called when the Rewards panel has been closed.
+    virtual void OnRewardsPanelClosed() {}
+  };
```

Comment:
I'd use a `struct` here to save a few characters:
```
struct Observer : base::CheckedObserver {
  virtual void OnRewardsPanelRequested(const mojom::RewardsPanelArgs& args) {}
  virtual void OnRewardsPanelClosed() {}
};
```

Reply:
I would like this change, but I think the general convention for these kinds of things to be `class`.

Reply:
Yeah, just checked, [this](https://google.github.io/styleguide/cppguide.html#Structs_vs._Classes) guideline is a bit pedantic.